onnxruntime/core/graph/graph.cc

```diff
@@ -1310,7 +1310,8 @@ Graph::Graph(const Model& owning_model,
   }
 
   for (const auto& node_proto : graph_proto_->node()) {
-    AddNode(node_proto, name_to_type_map);
+    auto& new_node = AddNode(node_proto, name_to_type_map);
+    SetOpSchemaFromRegistryForNode(new_node);
```

Comment:
If you do this here does that result in the checker never running for the node? i.e. do you improve performance but part of that is from not running the onnx validation which includes type/shape inferencing?

Kernel implementations can rely on that to have run - e.g. they don't duplicate the validation logic because they assume it has previously run.

Reply:
Sorry - type/shape inferencing is separate, but the checker is validating the node conforms to the schema. Kernel implementations also depend on that, for example by assuming there will be the correct number of required inputs.

Reply:
Seems like this line is a breaking change. However this PR still makes sense without it. An alternative could also be `try`-ing `checker::check_node` with default parameters in `Graph::Graph`, but this could lead to regressions. So I will change the PR to remove this line instead.